embassy-mspm0/src/wwdt.rs

```diff
@@ -0,0 +1,345 @@
+            | Self::MSec380
+            | Self::Sec3
+            | Self::Sec24
+            | Self::Sec192
+            | Self::Sec3072 => 2u8,
+            //  divide by 4
+            Self::USec32250
+            | Self::USec125000
+            | Self::MSec500
+            | Self::Sec4
+            | Self::Sec32
+            | Self::Sec256
+            | Self::Sec4096 => 3u8,
+            //  divide by 5
+            Self::USec9770
+            | Self::USec39060
+            | Self::USec156250
+            | Self::MSec630
+            | Self::Sec5
+            | Self::Sec40
+            | Self::Sec320
+            | Self::Sec5120 => 4u8,
+            //  divide by 6
+            Self::USec11720
+            | Self::USec46880
+            | Self::USec187500
+            | Self::MSec750
+            | Self::Sec6
+            | Self::Sec48
+            | Self::Sec384
+            | Self::Sec6144 => 5u8,
+            //  divide by 7
+            Self::USec13670
+            | Self::USec54690
+            | Self::USec218750
+            | Self::MSec880
+            | Self::Sec7
+            | Self::Sec56
+            | Self::Sec448
+            | Self::Sec7168 => 6u8,
+            //  divide by 8
+            Self::Sec512 | Self::Sec8192 => 7u8,
+        }
+    }
+}
+
+/// Timeout percentage that is treated as "too early" and generates violation.
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub enum ClosedWindowPercentage {
+    // window period is not used
+    Zero,
+    // 12.5% percents
+    Twelve,
+    // 18.75% percents
+    Eighteen,
+    // 25% percents
+    TwentyFive,
+    // 50% percents
+    Fifty,
+    // 75% percents
+    SeventyFive,
+    // 81.25% percents
+    EightyOne,
+    // 87.5% percents
+    EightySeven,
+}
+
+impl ClosedWindowPercentage {
+    fn get_native_size(self) -> vals::Window {
+        match self {
+            Self::Zero => vals::Window::SIZE_0,
+            Self::Twelve => vals::Window::SIZE_12,
+            Self::Eighteen => vals::Window::SIZE_18,
+            Self::TwentyFive => vals::Window::SIZE_25,
+            Self::Fifty => vals::Window::SIZE_50,
+            Self::SeventyFive => vals::Window::SIZE_75,
+            Self::EightyOne => vals::Window::SIZE_81,
+            Self::EightySeven => vals::Window::SIZE_87,
+        }
+    }
+}
+
+#[non_exhaustive]
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+/// Watchdog Config
+pub struct Config {
+    /// Watchdog timeout
+    pub timeout: Timeout,
+
+    /// closed window percentage
+    pub closed_window: ClosedWindowPercentage,
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Self {
+            timeout: Timeout::Sec1,
+            closed_window: ClosedWindowPercentage::Zero,
+        }
+    }
+}
+
+pub struct Watchdog {
+    regs: &'static Regs,
+}
+
+impl Watchdog {
+    /// Watchdog initialization.
+    pub fn new<T: Instance>(_instance: Peri<T>, config: Config) -> Self {
```

Comment:
I was confused why you were having `new` take ownership of the peripheral, but this is apparently correct per reference manual:

> The first write (with a key match) to the WWDTCTL0 register enables the WWDT. Once the WWDT is enabled, the WWDTCTL0 register becomes write protected.